packages/vscode-extension/src/webview/components/NetworkBar.tsx

```diff
@@ -0,0 +1,51 @@
+import IconButton from "./shared/IconButton";
+import { useNetwork } from "../providers/NetworkProvider";
+
+const NetworkBar = () => {
+  const { isRecording, toggleRecording, clearActivity, toggleShowSearch, toggleShowChart } =
+    useNetwork();
+
+  return (
+    <>
+      <IconButton
+        onClick={toggleRecording}
+        tooltip={{
+          label: isRecording
+            ? "Stop recording network activity"
+            : "Start recording network activity",
+          side: "bottom",
+        }}>
+        <span
+          style={{ color: isRecording ? "var(--vscode-charts-red)" : "var(--swm-default-text)" }}
+          className={`codicon ${isRecording ? "codicon-record" : "codicon-stop-circle"}`}
```

Comment:
might be a bit easier to read using [classnames](https://www.npmjs.com/package/classnames) - we're already using in the project